src/app/dataset-view/additional-components/metadata-component/components/form-components/select-date-format-field/select-date-format-field.component.ts

```diff
@@ -44,10 +44,14 @@ export class SelectDateFormatFieldComponent extends BaseField implements OnInit
         const subscription = this.eventTimeGroup
             .get(this.KIND_NAME_CONTROL)
             ?.valueChanges.subscribe((kind: EventTimeSourceKind) => {
-                if (kind === this.eventTimeSourceKind.FROM_METADATA) {
-                    this.currentSource = EventTimeSourceKind.FROM_METADATA;
+                if (kind !== EventTimeSourceKind.FROM_PATH) {
                     this.eventTimeGroup.removeControl(this.PATTERN_NAME_CONTROL);
                     this.eventTimeGroup.removeControl(this.TIMESTAMP_FORMAT_NAME_CONTROL);
+                }
+                if (kind === EventTimeSourceKind.FROM_METADATA) {
+                    this.currentSource = EventTimeSourceKind.FROM_METADATA;
+                } else if (kind === EventTimeSourceKind.FROM_SYSTEM_TIME) {
+                    this.currentSource = EventTimeSourceKind.FROM_SYSTEM_TIME;
                 } else {
                     this.currentSource = EventTimeSourceKind.FROM_PATH;
```

Comment:
Just assign `this.currentSource = kind` unconditionally

Reply:
Fixed